File: 0792-binary-search/0792-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int L = 0, R = nums.size() - 1;

        while (L <= R){
            int mid = L + (R - L) / 2; // To avoid overflow of integers
            int guess = nums[mid];

            if (guess == target)
                return mid;
            else if (guess > target)
                R = mid - 1;
            else 
                L = mid + 1;
        }
        return -1;
    }
};

File: 0792-binary-search/0792-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        L, R = 0, len(nums) - 1

        while L <= R:
            mid = L + (R - L) // 2 # to avoid overflow
            guess = nums[mid]
            if guess == target:
                return mid
            elif guess > target:
                R = mid - 1
            else:
                L = mid + 1
        return -1
        